aepsych/server/server.py

```diff
@@ -61,21 +63,38 @@ def __init__(self, socket=None, database_path=None, thrift=False):
 
         self.debug = False
         self.is_using_thrift = thrift
+        self.receive_thread = threading.Thread(
+            target=self._receive_send, args=(self.exit_server_loop,), daemon=True)
+
+        self.queue = []
 
     def cleanup(self):
         self.socket.close()
 
-    def _receive_send(self):
-        request = self.socket.receive()
-        try:
-            if "version" in request.keys():
-                result = self.versioned_handler(request)
+    def _receive_send(self, is_exiting):
+        while True:
+            request = self.socket.receive(is_exiting)
+            if request != 'bad request':
+                self.queue.append(request)
             else:
-                result = self.unversioned_handler(request)
-        except Exception as e:
-            result = "bad request"
-            logger.warning(f"Request '{request}' raised error '{e}'")
-
+                logger.info(
```

Comment:
I think this message is redundant because you already log when you catch the bad message, right?

Reply:
This is correct, I forgot that message execution errors were handled in the request function.